src/main/java/slimeknights/tconstruct/library/modifiers/modules/technical/MaxArmorLevelModule.java

```diff
@@ -50,12 +46,12 @@ public List<ModuleHook<?>> getDefaultHooks() {
 
   @Override
   public void onEquip(IToolStackView tool, ModifierEntry modifier, EquipmentChangeContext context) {
-    findMaxLevelIfArmor(tool, context, key, modifier.intEffectiveLevel(), allowBroken, heldTag);
+    updateMaxLevelIfArmor(tool, context, key, modifier.intEffectiveLevel(), allowBroken, heldTag);
   }
 
   @Override
   public void onUnequip(IToolStackView tool, ModifierEntry modifier, EquipmentChangeContext context) {
-    findMaxLevelIfArmor(tool, context, key, -modifier.intEffectiveLevel(), allowBroken, heldTag);
+    updateMaxLevelIfArmor(tool, context, key, -modifier.intEffectiveLevel(), allowBroken, heldTag);
   }
```

Comment:
Thinking about this more while I was attempting to fall asleep, I think this won't work. Your logic on equip is great, but on unequip the max will always be what was previously equip (as unlike with add, max is not inverted by a negative).

We thus have two options:
1. Bring back what I did with armor max levels before. There should be an object you can use that sets a max level per slot.
2. On unequip, iterate all 6 slots to find the new max level.

Its a question of whether we want efficiency of memory or efficiency of code. My old solution is arguably better just as it scales better, modifier lookup is a loop over a potentially large set even if the slots are a small loop. Either case probably won't play nicely with multiple modifiers using the same key (you will get the max per modifier rather than the sum of both modifiers); if you end up doing the object key then just make this unserializable for now until we need it serializable.